eventually/authentication/models.py

```diff
@@ -0,0 +1,133 @@
+"""
+This file contains models for the postgresql database.
+"""
+
+from datetime import datetime
+from django.db import models
+from django.contrib.auth.base_user import AbstractBaseUser
+from django.utils.dateformat import format
+
+class CustomUser(AbstractBaseUser):
+    """
+        This class represents a basic user. \n
+        1. first, last and middle names: all Char with maximum length of 20 chars.
+        2. email: email field of max length 20 chars
+        3. password: charfield of max length 128 symbols
+        4. updated_at & created_at fields: DateTimeField.
+           created_at can't be changed once created
+    """
+
+    first_name = models.CharField('first_name', max_length=20)
+    middle_name = models.CharField('middle_name', max_length=20)
+    last_name = models.CharField('second_name', max_length=20)
+    email = models.EmailField('email_address', max_length=40, unique=True)
+    password = models.CharField('password', max_length=128)
+    updated_at = models.DateTimeField(auto_now=True)
+    created_at = models.DateTimeField(auto_now_add=True, editable=False)
+
+    def __str__(self):
+        """
+            This magic method is redefined to show first, last name and id of the CustomUser object.
+        """
+        return '{} {} {}'.format(self.first_name, self.last_name, self.id)
+
+    @staticmethod
+    def get_by_id(user_id):
+        """
+        :param user_id: the id of a user to be found in the DB
+        :return: user object or None if a user with such ID does not exist
+        """
+        try:
+            return CustomUser.objects.get(id=user_id)
+        except CustomUser.DoesNotExist:
+            return None
+
+    @staticmethod
+    def delete_by_id(user_id):
+        """
+        :param user_id: an id of a user to be deleted
+        :return: True if object existed in the db and was removed or False if it didn't exist
+        """
+        try:
+            user = CustomUser.objects.get(id=user_id)
+            user.delete()
+            return True
+        except AttributeError:
+            return False
```

Comment:
from logging import getLogger
log = getLogger('log.ManageUsers')
log.warn('Cannot delete user {}. Please check correctnes of your request.')

Reply:
Where does this code has to be applied?

Reply:
instead of return 'False'. In every place where you have any error or ...

Reply:
Logger added.